packages/connect-form/src/components/advanced-options-tabs/advanced-tab/url-options-table.tsx

```diff
@@ -0,0 +1,251 @@
+      updateConnectionFormField({
+        type: 'update-search-param',
+        // If the user is selecting key for the first time, currentName is undefined and we choose the selected value.
+        currentKey: currentName ?? name,
+        // If the keys are same, then user is changing value. So we set newKey to undefined.
+        newKey: currentName === name ? undefined : name,
+        value: value ?? '',
+      });
+    }
+  };
+
+  const deleteUrlOption = (name?: UrlOption['name']) => {
+    setErrorMessge('');
+    const indexOfDeletedOption = options.findIndex(
+      (option) => option.name === name
+    );
+    const newOptions = [...options];
+    newOptions.splice(indexOfDeletedOption, 1);
+    setOptions(newOptions);
+    if (name) {
+      updateConnectionFormField({
+        type: 'delete-search-param',
+        key: name,
+      });
+    }
+  };
+
+  return (
+    <>
+      <Table
+        data-testid="url-options-table"
+        data={options}
+        columns={[
+          <TableHeader
+            key={'name'}
+            label="Key"
+            className={tableHeaderStyles}
+          />,
+          <TableHeader
+            key={'value'}
+            label="Value"
+            className={tableHeaderStyles}
+          />,
+        ]}
+      >
+        {({ datum }: { datum: Partial<UrlOption> }) => (
+          <Row
+            key={datum.name}
+            data-testid={
+              datum.name ? `${datum.name}-table-row` : 'new-option-table-row'
+            }
+          >
+            <Cell>
+              <Select
+                placeholder="Select key"
+                name="name"
+                aria-labelledby={
+                  datum.name ? `${datum.name} select` : 'new option select'
+                }
+                onChange={(name, event): void => {
+                  event.preventDefault();
+                  updateUrlOption(
+                    datum.name,
+                    name as UrlOption['name'],
+                    datum.value
+                  );
+                }}
+                allowDeselect={false}
+                value={datum.name ?? ''}
+              >
+                {editableUrlOptions.map(({ title, values }) => (
+                  <OptionGroup key={title} label={title}>
+                    {values.map((value) => (
+                      <Option
+                        key={value}
+                        value={value}
+                        // Disable if this option already exists in search params
+                        disabled={
+                          connectionStringUrl.searchParams.has(value) &&
+                          datum.name !== value
+                        }
+                      >
+                        {value}
+                      </Option>
+                    ))}
+                  </OptionGroup>
+                ))}
+              </Select>
+            </Cell>
+            <Cell>
+              <div className={optionValueCellStyles}>
```

Comment:
Definitely a small thing, could we instead pass a `className` to the `Cell` and not have a div? Looks like it takes that prop https://www.mongodb.design/component/table/documentation/

Reply:
yeah, but as we discussed `Cell` wraps the content in a `span` tag and then we will have to style that tag which I don't find clean. I had done that, but ended up doing this. (also @gribnoysup suggested this).

Reply:
Ah sgtm